### DISCUSSION TITLE
[Suggestion] Visual display of basket contents

### DISCUSSION THREAD

**Discussion by MarioSMB** [vectorwing/FarmersDelight#530] [Ideas]
The basket is a nice decorative storage block, but it seems a little strange to be able to see through and inside it and not see any items in it when it's full!  
It would be a nice touch to display states of full-ness in a similar manner to compost bins, so it looks like it has a bunch of random items thrown in.

**Comment by vectorwing** [comment]
I've thought about this in the past, but ended up not committing to it for a few reasons:
- Hoppers, though shallow, don't display fullness of items in their model;
- Making a representative "clutter" texture didn't quite work out, and rendering many item stacks inside of it looked noisy;
- Baskets can be upside down; I feel that rendering stuff inside would break suspension of disbelief of things not spilling out. 😅

Regardless, something to consider trying again sometime.

**Comment by MarioSMB** [comment]
Those points are fair, though I would argue that hoppers have an intentionally darkened interior to match the design of piping items through it with minimal storage, while baskets are completely hollow!  
An alternative might be to make the insides of the basket a simple flat texture similar to that of chests, so baskets would appear to be holding contents while not actually needing to render anything within.

**Comment by Aceplante** [comment]
Support, this would be cool if you can figure it out
